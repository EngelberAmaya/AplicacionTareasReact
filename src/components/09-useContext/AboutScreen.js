import React from 'react'
import { UserContext } from './UserContext';

const AboutScreen = () => {

    const {user, setUser} = React.useContext(UserContext);

    const handleClick = () => {
        setUser({})
    }

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>

            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>

            <button onClick={handleClick} className="btn btn-primary">LogOut</button>
        </div>
    )
}

export default AboutScreen
