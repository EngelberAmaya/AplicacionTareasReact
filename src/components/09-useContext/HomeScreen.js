import React from 'react'
import { UserContext } from './UserContext';

const HomeScreen = () => {

    const userContext = React.useContext(UserContext);
    console.log(userContext);

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr/>
        </div>
    )
}

export default HomeScreen
