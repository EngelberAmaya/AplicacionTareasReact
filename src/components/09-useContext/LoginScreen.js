import React from 'react'
import { UserContext } from './UserContext';

const LoginScreen = () => {

    const {setUser} = React.useContext(UserContext);

    const data = {
        id: 123,
        name: 'Engelber'
    }

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button onClick={() => setUser(data)} className='btn btn-primary'>Login</button>
        </div>
    )
}

export default LoginScreen
