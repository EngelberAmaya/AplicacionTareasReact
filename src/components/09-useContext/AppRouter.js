import React from 'react';
import { BrowserRouter, Routes, Route } from "react-router-dom";
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import Navbar from './Navbar';

const AppRouter = () => {

    return (
        <BrowserRouter>
            <Navbar/>
            <Routes>
                <Route path='/' element={<HomeScreen />} />                
                <Route path='/about' element={<AboutScreen />} />
                <Route path='/login' element={<LoginScreen />} />
                <Route path="*" element={<HomeScreen />} />
            </Routes>
        </BrowserRouter>
    )
}

export default AppRouter
