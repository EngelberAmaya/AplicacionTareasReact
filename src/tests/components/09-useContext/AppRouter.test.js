import { UserContext } from "../../../components/09-useContext/UserContext";
import {mount} from 'enzyme';
import AppRouter from "../../../components/09-useContext/AppRouter";

describe('Pruebas en <AppRouter />', () => {

   const user = {
      id: 1,
      name: 'Engelber'
   }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter />
        </UserContext.Provider>
    );    

    test('Debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
      
    });
  
})
