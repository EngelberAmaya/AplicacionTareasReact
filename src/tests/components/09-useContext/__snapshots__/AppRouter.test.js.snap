// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el componente correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          activeclassname="active"
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          activeclassname="active"
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          activeclassname="active"
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Engelber"
}
      </pre>
    </div>
  </div>,
]
`;
